library/common/extensions/filters/http/platform_bridge/c_type_definitions.h

```diff
@@ -13,15 +13,34 @@ const envoy_filter_headers_status_t kEnvoyFilterHeadersStatusContinueAndEndStrea
 const envoy_filter_headers_status_t kEnvoyFilterHeadersStatusStopAllIterationAndBuffer =
     static_cast<envoy_filter_headers_status_t>(
         Envoy::Http::FilterHeadersStatus::StopAllIterationAndBuffer);
+// ResumeIteration is not a status supported by Envoy itself, and only has relevance in Envoy
+// Mobile's implementation of platform filters.
+//
+// Regarding enum values, the C++11 standard (7.2/2) states:
+//   If the first enumerator has no initializer, the value of the corresponding constant is zero.
+//   An enumerator-definition without an initializer gives the enumerator the value obtained by
+//   increasing the value of the previous enumerator by one.
+//
+// Creating a new return status like this is brittle, but at least somewhat more resilient to
+// a new status being added in Envoy, since it won't overlap as long as the new status is added
+// rather than prepended.
+const envoy_filter_headers_status_t kEnvoyFilterHeadersStatusResumeIteration =
```

Comment:
After looking at onData, I am still confused why we need two different statuses rather than driving difference via the presence of absence of headers to replace (or headers/data in onTrailers). I might just be being dense, but I wonder if we can eliminate the need of another status to support the use case?

If we _do_ need the new status, can we write more documentation about it in the filter.cc file?

Reply:
I _think_ I see? In continue iteration the buffer copied is only the current frame, but in resume iteration it is the entire internal buffer?

Reply:
@junr03 I think it will be helpful for you to reference the API doc we put together back when we discussed our implementation for advanced filters. It lays out differences between Envoy's native filters and the platform filters. Also the docstrings on the platform implementations of these APIs contain some detail as well, e.g.
https://github.com/lyft/envoy-mobile/blob/main/library/swift/src/filters/AsyncRequestFilter.swift
https://github.com/lyft/envoy-mobile/blob/main/library/swift/src/filters/FilterResumeStatus.swift

Reply:
Sorry actually that's the other side of the coin - I mixed up the PRs because I'm working on the asynchronous side right now. For the synchronous side there's this (e.g.):
https://github.com/lyft/envoy-mobile/blob/main/library/swift/src/filters/FilterDataStatus.swift
Because iteration may have been stopped on headers, we need a mechanism for the platform filter to still inform Envoy of what modifications, if any, it wants to perform on headers when iteration is resumed. Same for headers _and_ data, when resumption occurs on trailers.

Reply:
Part of the reason to have two statuses is that it keeps things simple for the main/golden path. When all you're doing is continuing, you never need to worry about sending previous entities.